src/package-request.js

```
@@ -134,14 +134,20 @@ export default class PackageRequest {
 
   async normalizeRange(pattern: string): Promise<string> {
     if (pattern.indexOf(':') > -1 || pattern.indexOf('@') > -1 || getExoticResolver(pattern)) {
-      return Promise.resolve(pattern);
+      return pattern;
     }
 
-    if (await fs.exists(path.join(this.config.cwd, pattern))) {
-      return Promise.resolve(`file:${pattern}`);
+    if (!semver.validRange(pattern)) {
+      try {
+        if ((await fs.stat(path.join(this.config.cwd, pattern))).isDirectory()) {
```

That being said, I checked on NPM and it *seems* that they just silently ignore `"foo": "foo.tgz"` ... so I guess this PR should be fine :)